src/zql/ivm/view/tree-view.ts

```diff
@@ -65,7 +65,7 @@ class AbstractTreeView<T> extends AbstractView<T, T[]> {
 
     let newData = this.#data;
     for (const c of collections) {
-      [changed, newData] = this.#sink(c, newData) || changed;
+      [changed, newData] = this.#sink(c, newData, changed);
```

Comment:
I was stepping through the code and found this strange code.

`this.#sink` never returns a falsey value.

It seems to me based on looking at the code that changed could go from true to false if some later collection does not change anything.